sentry_sdk/integrations/chalice.py

```diff
@@ -0,0 +1,93 @@
+import sys
+
+from functools import wraps
+import chalice
+from chalice import Chalice, ChaliceViewError
+from chalice.app import EventSourceHandler as ChaliceEventSourceHandler
+from sentry_sdk._compat import reraise
+from sentry_sdk.hub import Hub
+from sentry_sdk.integrations import Integration
+from sentry_sdk.integrations.aws_lambda import _make_request_event_processor
+from sentry_sdk.utils import (
+    capture_internal_exceptions,
+    event_from_exception,
+)
+
+
+class EventSourceHandler(ChaliceEventSourceHandler):
+    def __call__(self, event, context):
+        hub = Hub.current
+        client = hub.client
+
+        with hub.push_scope() as scope:
+            with capture_internal_exceptions():
+                configured_time = context.get_remaining_time_in_millis()
+                scope.add_event_processor(
+                    _make_request_event_processor(event, context,
+                                                  configured_time)
+                )
+            try:
+                event_obj = self.event_class(event, context)
+                return self.func(event_obj)
+            except Exception:
+                exc_info = sys.exc_info()
+                event, hint = event_from_exception(
+                    exc_info,
+                    client_options=client.options,
+                    mechanism={"type": "chalice", "handled": False},
+                )
+                hub.capture_event(event, hint=hint)
+                hub.flush()
+                reraise(*exc_info)
+
+
+old_get_view_function_response = Chalice._get_view_function_response
```

Comment:
Please don't have a global backup of the view function response, this will break if multiple libraries monkeypatch chalice

Reply:
add this in setup_once().

Reply:
Regarding the tests, these fail because the parameters of the context '_max_runtime' arrive null, so this line is not executed, the response if it contains the status_code '500' and the json, I don't understand why it fails in travis.